### ISSUE TITLE
Upgrade to moodle 3.2 fails

### ISSUE THREAD

**Issue by JosiahCarberry** [CTANZ/moodle-theme_decaf#28]
I have decaf as my theme and found that the upgrade to Moodle 3.2 fails, when that same upgrade succeeds if I first change the theme to clean.

During the upgrade procedure, all the plugins are upgraded without error. Then, after clicking on Continue, the following error message is displayed:
Click on Continue

The following error messages displayed:

Notice: Undefined index: mycourses in /home/clients/bc98c1b3fe626afdda48129babd460b2/web/training/lib/navigationlib.php on line 

2522
Exception - Call to a member function find() on null
Debug info:
Error code: generalexceptionmessage
Stack trace:

    line 2522 of /lib/navigationlib.php: Error thrown
    line 683 of /theme/decaf/lib.php: call to global_navigation->add_course()
    line 337 of /theme/decaf/lib.php: call to theme_decaf_expand_navigation->add_course()
    line 319 of /theme/decaf/lib.php: call to theme_decaf_expand_navigation->initialise()
    line 88 of /theme/decaf/classes/topsettings_renderer.php: call to theme_decaf_expand_navigation->__construct()
    line 21 of /theme/decaf/classes/topsettings_renderer.php: call to theme_decaf_topsettings_renderer->navigation_node()
    line 49 of /theme/decaf/lib.php: call to theme_decaf_topsettings_renderer->navigation_tree()
    line 18 of /theme/decaf/layout/columns1.php: call to theme_decaf_init()
    line 1110 of /lib/outputrenderers.php: call to include()
    line 1040 of /lib/outputrenderers.php: call to core_renderer->render_page_layout()
    line 1827 of /lib/setuplib.php: call to core_renderer->header()
    line 342 of /login/index.php: call to bootstrap_renderer->__call()

**Comment by mwehr** [comment]
Hi,

in lib.php, moving:
 ```php
$this->rootnodes['mycourses'] = $this->add(get_string('mycourses'), new moodle_url('/my'), self::TYPE_ROOTNODE, null, 'mycourses');
```
under 
```php
$this->rootnodes = array();
```
should do the job
```php
    public function initialise() {
        global $CFG, $DB, $SITE, $PAGE;

        if ($this->initialised || during_initial_install()) {
            return $this->expandable;
        }
        $this->initialised = true;

        $this->rootnodes = array();
        $this->rootnodes['mycourses'] = $this->add(get_string('mycourses'), new moodle_url('/my'), self::TYPE_ROOTNODE, null, 'mycourses');
        $this->rootnodes['site']      = $this->add_course($SITE);
        $this->rootnodes['currentcourse'] = $this->add(get_string('currentcourse'), null, self::TYPE_ROOTNODE, null, 'currentcourse');
        $this->rootnodes['courses'] = $this->add(get_string('courses'), null, self::TYPE_ROOTNODE, null, 'courses');
```

**Comment by gjb2048** [comment]
I moved:
$this->rootnodes['site']      = $this->add_course($SITE);
to the bottom and that did the job.

**Comment by pauln** [comment]
Fixed in v2.2.0 (just released).  Thanks to @gjb2048 for submitting a pull request!
